### ISSUE TITLE
Sorting a large number of content causes a time-out error

### ISSUE THREAD

**Issue by vonbv** [umbraco/Umbraco-CMS#3936]
We have almost 500 content. Creating a new one puts the new content as the last item under a parent node (sort order = max number in the list). When we sort the child nodes by dragging the item to the top it takes a very long time until the "UpdateSortOrder" results into a 500 error with no details of the error.

Reproduction
------------

Right-click a node, select "Sort", drag the last item to the top (list should be long, we have not tested with less than 100, ours has child nodes of about 450-500, hit Save.

**Comment by emmaburstow** [comment]
Thanks for this report @vonbv - we'll try to reproduce and then set it as up for grabs if we can. Are you able to tell us which version you're running in the meantime?

Thanks,

Emma

**Comment by nul800sebastiaan** [comment]
I'll close this for now as I've also not be able to reproduce in 7.12.4. If you have more information then we're all ears @vonbv. Also take into consideration that you might have custom code running that responds to sorting events?

**Comment by vonbv** [comment]
Sorry I was not notified of your replies. We are using 7.12.3. We do not have codes that are subscribed to any Umbraco events.

**Comment by nul800sebastiaan** [comment]
Thanks for the reply! Unfortunately I can't reproduce this at all with about 600 nodes in a tree, there may be some performance problems on your end that you could look into, maybe SQL lag or a server that's under high load? Make sure to create an post on the forum to get some more help with this!

**Comment by vonbv** [comment]
Okay thanks for looking @nul800sebastiaan

**Comment by vonbv** [comment]
I saw where the problem was and part of it is because the database is in another region far from the web app server. However, I noticed something that can be improve. In the Commit method in ScopeUnitOfWork the _operation.Count is double of the "item count being sorted". There is always an _insert_ and an _update_ per sorted item. In our case, when we moved an item from the bottom to the top of a 400-item list, the count of sorted item is 400. Then in the Commit method the _operations.Count is doubled (800), see the attached images.
![image](https://user-images.githubusercontent.com/870921/51552930-b1efc600-1eac-11e9-972f-9d4f2a41e933.png)
![image](https://user-images.githubusercontent.com/870921/51552947-b6b47a00-1eac-11e9-895e-62a5aa0c6747.png)
